ompi/communicator/comm_init.c

```diff
@@ -169,6 +170,10 @@ int ompi_comm_init(void)
     (void)opal_pointer_array_test_and_set_item(&ompi_mpi_communicators, 0, &ompi_mpi_comm_null);
     (void)opal_pointer_array_test_and_set_item(&ompi_mpi_communicators, 1, &ompi_mpi_comm_null);
 
+    ompi_mpi_comm_null.comm.c_name = (char *) malloc (OPAL_MAX_OBJECT_NAME);
+    if (OPAL_UNLIKELY(NULL == ompi_mpi_comm_null.comm.c_name)) {
+        return OMPI_ERR_OUT_OF_RESOURCE;
+    }
     opal_string_copy(ompi_mpi_comm_null.comm.c_name, "MPI_COMM_NULL",
                      sizeof(ompi_mpi_comm_null.comm.c_name));
```

Comment:
This won't work: `sizeof(ompi_mpi_comm_null.comm.c_name)` won't return the array size anymore. Why not just call strdup?

Reply:
argh, you are right. I had it in another branch correctly set, but forgot to transfer that part to this branch. Will fix it. Thanks!